custom_model_runner/datarobot_drum/drum/gpu_predictors/triton_predictor.py

```diff
@@ -95,8 +98,29 @@ def has_read_input_data_hook(self):
     def _predict(self, **kwargs) -> RawPredictResponse:
         raise DrumCommonException("Predict structured is not supported")
 
+    def _get_inference_header_size(self, data):
+        try:
+            first_line = io.BytesIO(data).readline()
+            json.loads(first_line)  # read first line to check it's a valid json string
+            return len(first_line)
+        except Exception:
+            self.logger.warning("Can't calculate the inference header size", exc_info=True)
+            # perhaps inference header is not set into payload and request
+            # to Triton still may succeed
+            return  # do nothing
+
     def predict_unstructured(self, data, **kwargs):
-        headers = kwargs.get(UnstructuredDtoKeys.HEADERS)
+        headers = kwargs.get(UnstructuredDtoKeys.HEADERS, {})
+
+        # Predictions API does not forward the headers,
+        # while Triton expects to get inference header for the Binary Tensor request. See
+        # https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/protocol/extension_binary_data.html
+        if INFERENCE_HEADER.lower() not in set(k.lower() for k in headers):
+            inference_header_size = self._get_inference_header_size(data)
+            if inference_header_size:
+                headers.update({INFERENCE_HEADER: str(inference_header_size)})
+
+        self.logger.info(headers)
```

Comment:
```suggestion
        self.logger.debug(headers)
```
This looks more like debgging info?

Reply:
updated